db.go

```diff
@@ -528,6 +528,34 @@ func (db *DB) Close() error {
 	return merr.Err()
 }
 
+// DisableCompactions disables compactions.
+func (db *DB) DisableCompactions() error {
+	db.stopc <- struct{}{} // TODO: Can this block?
+	db.cmtx.Lock()
+	return nil
+}
+
+// EnableCompactions enables compactions.
+func (db *DB) EnableCompactions() error {
+	db.cmtx.Unlock()
+	return nil
+}
+
+// Snapshot writes the current headBlock snapshots to snapshots directory.
+func (db *DB) Snapshot(dir string) error {
+	db.headmtx.RLock()
+	heads := db.heads[:]
+	db.headmtx.RUnlock()
+
+	for _, h := range heads {
```

Comment:
So additionally to snapshotting head blocks into the directory. We should hard link all persisted blocks into the dir as well.

Did deletions on persisted blocks create a new directory or were they added to existing ones? If the latter, we either have to change that, or require or enforce that not deletions are done while snaphots are being shipped off.
Theoretically, compaction can be resumed right after Snapshot returns (an argument for putting the enable/disbale calls right into this method btw). So this is not guaranteed through `ctmx` alone.